### PR TITLE
Complying with quindy changes for message envelope refactor

### PR DESCRIPTION
**Pull request by pimotte**
Exactly what it says on the tin.

### CODE REVIEW

File: university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package nl.quintor.studybits.controller;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import nl.quintor.studybits.LedgerSeeder;
```
Comment (daneduijnkerke) [inline university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java]: Unused LedgerSeeder

File: university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java [hunk 2]
```diff
@@ -22,13 +23,17 @@
 
     @Autowired
     private ExchangePositionService exchangePositionService;
+
+    @Autowired
+    private LedgerSeeder ledgerSeeder;
```
Comment (daneduijnkerke) [inline university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java]: Unused field

File: university-agent/src/main/java/nl/quintor/studybits/controller/BootstrapController.java [hunk 3]
```diff
@@ -34,6 +32,9 @@
     @Autowired
     private Seeder seeder;
 
+    @Autowired
```
Comment (daneduijnkerke) [inline university-agent/src/main/java/nl/quintor/studybits/controller/BootstrapController.java]: Autowired should be optional (@Autowired(required = false))

File: university-agent/src/main/java/nl/quintor/studybits/LedgerSeeder.java [hunk 4]
```diff
@@ -89,7 +84,50 @@ public void seed(ContextRefreshedEvent event) throws InterruptedException, Execu
...
 
-    private boolean needsSeeding() {
+    public static void onboardIssuer(TrustAnchor steward, Issuer newcomer) throws InterruptedException, ExecutionException, IndyException, IOException {
+        // Create Codecs to facilitate encryption/decryption
+        MessageEnvelopeCodec stewardCodec = new MessageEnvelopeCodec(steward);
+        MessageEnvelopeCodec newcomerCodec = new MessageEnvelopeCodec(newcomer);
+
+        // Connecting newcomer with Steward
+
+        // #Step 4.1.2 & 4.1.3
```
Comment (daneduijnkerke) [inline university-agent/src/main/java/nl/quintor/studybits/LedgerSeeder.java]: Does this method need the step comments (used in quindy)?

File: university-agent/src/main/java/nl/quintor/studybits/service/AgentService.java [hunk 5]
```diff
@@ -23,46 +20,41 @@
...
+
     public MessageEnvelope processMessage(MessageEnvelope messageEnvelope) throws IndyException, ExecutionException, InterruptedException, IOException {
-        String messageTypeURN = messageEnvelope.getType().getURN();
+        String messageTypeURN = messageEnvelope.getMessageType().getURN();
 
-        messageEnvelope.setIndyWallet(universityIssuer);
 
         if (messageTypeURN.equals(CONNECTION_RESPONSE.getURN())) {
-            return handleConnectionResponse(messageEnvelope);
+            return handleConnectionResponse(MessageEnvelope.convertEnvelope(messageEnvelope, CONNECTION_RESPONSE));
```
Comment (daneduijnkerke) [inline university-agent/src/main/java/nl/quintor/studybits/service/AgentService.java]: Shouldn't you convert the envelop on CREDENTIAL_REQUEST and PROOF as well?

Comment (pimotte) [reply university-agent/src/main/java/nl/quintor/studybits/service/AgentService.java]: Done

File: university-agent/src/main/java/nl/quintor/studybits/LedgerSeeder.java (hunk 4 above)
Comment (pimotte) [reply university-agent/src/main/java/nl/quintor/studybits/LedgerSeeder.java]: Removed

File: university-agent/src/main/java/nl/quintor/studybits/controller/BootstrapController.java (hunk 3 above)
Comment (pimotte) [reply university-agent/src/main/java/nl/quintor/studybits/controller/BootstrapController.java]: Done

File: university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java (hunk 2 above)
Comment (pimotte) [reply university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java]: Removed

File: university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java (hunk 1 above)
Comment (pimotte) [reply university-agent/src/main/java/nl/quintor/studybits/controller/AgentController.java]: Removed

**Pull request merged**